File: Preview/go/listbuilds/listbuilds.go
package main

import (
	"context"
	"flag"
	"fmt"
	"log"
	"os"

	"cloud.google.com/go/cloudbuild/apiv1"
	"cloud.google.com/go/compute/metadata"
	"google.golang.org/api/iterator"
	cloudbuildpb "google.golang.org/genproto/googleapis/devtools/cloudbuild/v1"
)

func main() {
	ctx := context.Background()

	n := flag.Uint("limit", 10, "number of builds to display")
	flag.Parse()

	projectID, err := metadata.NewClient(nil).ProjectID()
	if err != nil {
		log.Fatalf("Unable to discern project: %v", err)
	}

	c, err := cloudbuild.NewClient(ctx)
	if err != nil {
		log.Fatalf("Unable to instantiate cloudbuild client: %v", err)
	}
	defer func() {
		if err := c.Close(); err != nil {
			log.Fatalf("Close() failed: %v", err)
		}
	}()

	req := &cloudbuildpb.ListBuildsRequest{ProjectId: projectID}
	it := c.ListBuilds(ctx, req)
	var i uint = 0
	for ; i < *n; i++ {
		b, err := it.Next()
		if err == iterator.Done {
			break
		}
		if err != nil {
			log.Fatalf("ListBuilds failure: %v", err)
		}
		fmt.Println("Build", b.Id, b.Status)
	}

	fmt.Printf("Listed %d builds for project %q.\n", i, projectID)
	os.Exit(0)
}

File: Preview/python/listbuilds/listbuilds.py
import sys

from google import auth
from google.cloud.devtools.cloudbuild_v1.services.cloud_build import client

def listbuilds(project_id):
    cb = client.CloudBuildClient()
    builds = cb.list_builds(project_id=project_id)

    i = 0
    for b in builds:
        print("Build", b.id, b.status.name)
        i += 1
        if i == 10:
            break

    print("Listed {} builds for project {}.".format(i, project_id))
    return


if __name__ == '__main__':
    _, project_id = auth.default()
    listbuilds(project_id)
